docs/updates/2018_04_02.md

```diff
@@ -0,0 +1,53 @@
+# Update for April 2, 2018
+
+## Contributions
+
+## Almost done with the file finder
+
+Our main focus last week was finishing up the file finder feature that I also [discussed in the previous update](./2018_03_26.md#fast-file-finding). The last update was all about our approach to scanning the directory tree from the file system into an in-memory representation, and the approach we described remains pretty much unchanged.
+
+### Leveraging prior art
+
+Last week was all about using that in-memory representation to return search results based on a "fuzzy" search query. After an initial attempt that yielded decent performance but poor ranking of the search results, we decided to investigate existing solutions. We tried two command-line fuzzy finders, [`fzy`](https://github.com/jhawthorn/fzy) (written in C) and [`fzf`](https://github.com/junegunn/fzf) (written in Go) on the Electron repository, which contains over 500,000 files when `.gitignore` is disabled.
```

Comment:
Do you know about [skim](https://github.com/lotabout/skim) which is a fuzzy finder written in Rust.

Reply:
Yeah, I ran across it in my initial search. It seemed to have a lot of features whereas `fzy` was pretty simple. If its approach offers a compelling advantage over the code we copied from `fzy` that would be good to know, although we still will probably just move on for now.